style <pre> elements

skpy opened the pull request:
Displaying code blocks in the `<pre><code>...</code></pre>` format would cause the entire fucking page to scroll.  By styling the `<pre>` element with `overflow: auto` we constrain the size of the block to the column, and scroll just within that fucking block.

I included some fucking padding, but feel free to strip that out.  Or feel free to add a fucking background color, to fucking disambiguate preformatted sections from the rest.

Commit:
style <pre> elements

KeenRivals commented:
Hmm, we don't have *any* ``<pre>`` blocks currently. This PR adds styles to ``<pre>``, so it doesn't actually do anything.

Commit:
who needs minifiers?

skpy commented:
I added a demonstration of the `<pre>` style inside the body of the page.

I also let my OCD get the better of me and organized the stylesheet elements by alphabetical order.

KeenRivals commented:
* I don't think  ``<code>`` needs to be wrapped with ``<pre>``, does it?
* Your editor is using spaces for indentation instead of tabs. 😒

I think you have a good idea to highlight the minimal stylesheet, but I'm not sure I want to embed it in the page or that I want to highlight the TCP window size.

In either case, the CSS rules to style ``<pre>`` elements are not needed because the page doesn't use them. I don't think the rules are needed for ``<code>`` elements either because the only code element I use currently is 7 characters long, so overflow is not a problem. Maybe it will be later, but right now it's not so I'll close this issue.

Thank you for opening this PR however, as it reminded me to consider ``<code>`` element overflow in a side project I'm working on (a very minimal Pelican theme for my personal website). Once I'm happy with it I'll have it here on GitHub.

skpy commented:
🤷‍♂️ Your project, your choice.

Most Markdown renderers will convert code fences (` ``` `) into `<pre><code>` and `</code></pre>` in order to preserve indentation within a multi-line code block.  A single backtick produces an inline `<code></code>` section not wrapped in `<pre></pre>`.  Without overflow control, the block will expand horizontally as long as the longest line, and trigger the entire page to scroll horizontally. That's not particularly awesome.

I added the text that I did so as not to have the stylesheet just hanging out there with no explanation.  I am rather proud of the snark I added about node.js minifiers, though. :)

Thanks for making the bestmotherfucking.website.  It's been helpful for me.

KeenRivals commented:
Thanks for the note on markdown renderers, that will help me out with my project. BMFW is handcrafted, *artisanal* html, so it doesn't really apply.

> I am rather proud of the snark I added about node.js minifiers

I appreciated that, having used ``cssnano`` for the first time recently I know what you mean. 😂

KeenRivals commented:
Thanks for the note on markdown renderers, that will help me out with my project. BMFW is handcrafted, *artisanal* html, so it doesn't really apply.

> I am rather proud of the snark I added about node.js minifiers

I appreciated that, having used ``cssnano`` for the first time recently I know what you mean. 😂
